prusti-specs/src/lib.rs

```diff
@@ -187,20 +197,18 @@ pub fn closure(tokens: TokenStream, drop_spec: bool) -> TokenStream {
         quote! {
             {
                 #cl_annotations #attrs_ts
-                let _prusti_closure =
```

Comment:
Why did you need to change this?

Reply:
Because `get_opt_spec_id()` in `prusti-viper/src/encoder/encoder.rs` receives the `DefId` of the function/closure and then checks the attributes associated with that `DefId` in the `TyCtxt`. In particular, this means that if we add the pre-/postcondition annotation to the assignment, as you suggest, then the encoder won't find the closure specification.

Reply:
`get_opt_spec_id` is intended for the top-level items. For example, it does not handle loop invariants. Unfortunately, I did not get around to fix them.

Reply:
Do you have a strong reason for this to get merged soon? Otherwise, I would prefer to wait until we fix support for the loop invariants, which should also solve your problem of attaching specs to closures.

Reply:
> Do you have a strong reason for this to get merged soon?

No, I just thought it would make sense to open a PR, given that I reached my first milestone (verification of closure bodies), to keep you updated and my branch from diverging too far.

On the other hand, given that "looking for the specification" is a somewhat unrelated problem from closure verification in particular, perhaps whoever fixes the loop invariants could also fix this issue while they're at it? I mean, it's not even "fixing", it's just replacing the `stmt_expr_attributes` solution with a different one.

Reply:
> No, I just thought it would make sense to open a PR, given that I reached my first milestone (verification of closure bodies), to keep you updated and my branch from diverging too far.

Having an open PR definitely helps for everyone to keep track of what is happening.

> On the other hand, given that "looking for the specification" is a somewhat unrelated problem from closure verification in particular, perhaps whoever fixes the loop invariants could also fix this issue while they're at it?

I think it will be much easier to fix the PR where we can clearly see the changed files than to clean it once the code is merged. Hopefully, this will not take too long.